urllib3/util/retry.py

```diff
@@ -169,10 +179,13 @@ def get_backoff_time(self):
 
         :rtype: float
         """
-        if self._observed_errors <= 1:
+        # We want to consider only the last consecutive errors sequence (Ignore redirects).
+        consecutive_errors_len = len(list(takewhile(lambda x: x.redirect_location is None,
```

Comment:
Er sorry I meant `consecutive_errors_len = next((i for i, h in enumerate(reversed(self.history)) if not h.redirect_location), 0)`

Fair enough.